File: scripts/test-connection.js
require("dotenv").config();

const pool = require("../src/config/db");

async function testConnection() {
  try {
    const result = await pool.query("SELECT NOW()");
    console.log("Connected to PostgreSQL.");
    console.log("Server time:", result.rows[0].now);
  } catch (error) {
    console.error("Failed to connect to PostgreSQL:", error.message);
    process.exitCode = 1;
  } finally {
    await pool.end();
  }
}

testConnection();

File: scripts/test_connection.py
import os
import sys
import asyncio
from dotenv import load_dotenv

sys.path.insert(0, os.path.abspath(os.path.join(os.path.dirname(__file__), '..')))

from src.config.db import get_pool, close_pool

async def main():
    load_dotenv()
    
    try:
        pool = await get_pool()
        
        async with pool.acquire() as conn:
            # Execute simple query
            server_time = await conn.fetchval('SELECT NOW();')
            
        print("Success: PostgreSQL/Neon is reachable.")
        print(f"Server timestamp: {server_time}")
    except Exception as e:
        print(f"Error connecting to database: {e}")
        sys.exit(1)
    finally:
        await close_pool()

if __name__ == "__main__":
    asyncio.run(main())
